File: android/src/main/kotlin/com/example/voip_twilio_sdk/VoipTwilioSdkPlugin.kt
package com.example.voip_twilio_sdk

import android.content.Context
import io.flutter.embedding.engine.plugins.FlutterPlugin

class VoipTwilioSdkPlugin : FlutterPlugin {
    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        val context = binding.applicationContext
        TwilioVoiceChannelHandler.getInstance().setup(binding.binaryMessenger, context)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        TwilioVoiceChannelHandler.getInstance().cleanup()
    }
}

File: ios/Classes/VoipTwilioSdkPlugin.swift
import Flutter
import UIKit

@objc public class VoipTwilioSdkPlugin: NSObject, FlutterPlugin {
    @objc public static func register(with registrar: FlutterPluginRegistrar) {
        TwilioVoiceChannelHandler.shared.setup(with: registrar.messenger())
    }
    
    public func detachFromEngine(for registrar: FlutterPluginRegistrar) {
        TwilioVoiceChannelHandler.shared.cleanup()
    }
}

